auto_remediate/security_hub_rules.py

```diff
@@ -75,6 +109,248 @@ def access_keys_rotated(self, resource_id):
+                f"Could not create new CloudWatch Log Group '{cloudwatch_log_group_name}'."
+            )
+            self.logging.error(sys.exc_info()[1])
+            return False
+        else:
+            try:
+                response = self.client_logs.describe_log_groups(
+                    logGroupNamePrefix=cloudwatch_log_group_name
+                )
+            except:
+                self.logging.error(
+                    f"Could not describe CloudWatch Log Group '{cloudwatch_log_group_name}'."
+                )
+                self.logging.error(sys.exc_info()[1])
+                self.delete_log_group(cloudwatch_log_group_name)
+                return False
+            else:
+                cloudwatch_log_group_arn = response.get("logGroups")[0].get("arn")
+                self.logging.info(
+                    f"Retrieved ARN '{cloudwatch_log_group_arn}' for CloudWatch Log Group '{cloudwatch_log_group_name}'."
+                )
+
+        # get trust relationship
+        try:
+            trust_relationship_file = "auto_remediate/data/cloud_trail_cloud_watch_logs_enabled_trust_relationship.json"
+            with open(trust_relationship_file) as file:
+                trust_relationship = str(file.read())
+        except:
+            self.logging.error(
+                f"Could not read IAM Trust Relationship file '{trust_relationship_file}'."
+            )
+            self.logging.error(sys.exc_info()[1])
+            self.delete_log_group(cloudwatch_log_group_name)
+            return False
+
+        # create IAM Role for CloudTrail
+        iam_role_name = f"CloudTrail-CloudWatchLogs-{resource_id}"
+        try:
+            response = self.client_iam.create_role(
+                RoleName=iam_role_name,
+                AssumeRolePolicyDocument=trust_relationship,
+                Description="AWS CloudTrail will assume the role to deliver CloudTrail events to the CloudWatch Logs log group",
+            )
+            self.logging.info(f"Created new IAM Role '{iam_role_name}'.")
+        except:
+            self.logging.error(f"Could not create new IAM Role '{iam_role_name}'.")
+            self.logging.error(sys.exc_info()[1])
+            self.delete_log_group(cloudwatch_log_group_name)
+            return False
+        else:
+            iam_role_arn = response.get("Role").get("Arn")
+
+            # create policy
+            try:
+                policy_file = "auto_remediate/data/cloud_trail_cloud_watch_logs_enabled_policy.json"
+                with open(policy_file) as file:
+                    policy = str(file.read())
+            except:
+                self.logging.error(f"Could not read IAM Policy file '{policy_file}'.")
+                self.logging.error(sys.exc_info()[1])
+                self.delete_log_group(cloudwatch_log_group_name)
+                return False
+            else:
+                policy = policy.replace("account_number", self.get_account_number())
```

Comment:
make the replacement targets more conspicuous perhaps like `__ACCOUNT_NUMBER__`
since we aren't going to go overboard templating like jsonnet

Reply:
45a00e8